crates/bevy_window/src/raw_handle.rs

```diff
@@ -44,17 +46,20 @@ unsafe impl Sync for RawHandleWrapper {}
 /// This can only be constructed via the [`RawHandleWrapper::get_handle()`] method;
 /// be sure to read the safety docs there about platform-specific limitations.
 /// In many cases, this should only be constructed on the main thread.
-pub struct ThreadLockedRawWindowHandleWrapper(RawHandleWrapper);
+pub struct ThreadLockedRawWindowHandleWrapper(pub RawHandleWrapper);
 
 // SAFETY: the caller has validated that this is a valid context to get [`RawHandleWrapper`]
 // as otherwise an instance of this type could not have been constructed
 // NOTE: we cannot simply impl HasRawWindowHandle for RawHandleWrapper,
 // as the `raw_window_handle` method is safe. We cannot guarantee that all calls
 // of this method are correct (as it may be off the main thread on an incompatible platform),
 // and so exposing a safe method to get a [`RawWindowHandle`] directly would be UB.
-unsafe impl HasRawWindowHandle for ThreadLockedRawWindowHandleWrapper {
-    fn raw_window_handle(&self) -> RawWindowHandle {
-        self.0.window_handle
+impl HasWindowHandle for ThreadLockedRawWindowHandleWrapper {
+    fn window_handle(&self) -> Result<WindowHandle, HandleError> {
+        // TODO: Unsure if this is the same safety as before
```

Comment:
I'll get this thread started since it's the biggest change here.

I believe we still need to use raw handles. I believe you cannot store objects with lifetimes in the world, at least not ones with a shorter lifetime. We may be able to transmute lifetimes to outlive the World, but that would give false safety.

Reply:
If you wrap `winit::window::Window` in an `Arc<Window>`, you can share it with the surface while getting a `'static` lifetime without any unsafe involved.